src/panels/lovelace/editor/card-editor/hui-edit-card.ts

```diff
@@ -264,7 +259,7 @@ export class HuiEditCard extends LitElement {
   }
 
   private _handleUIConfigChanged(value: LovelaceCardConfig): void {
-    this._configValue = { format: "json", value };
+    this._cardConfig = value;
```

Comment:
So when this happens, you need to reset the YAML . 

This is where the old approach was easier to reason about: there was only a single source of truth, and based on how it was going to be used, it was converted to the other format.

Now that we have 2 variables representing the same value, they need to be kept in sync.

Reply:
No, because this can't happen if the yaml is displayed, and when it isn't, `_cardYaml` is undefined so it will be regenerated the next time it's needed.

But I definitely see your point. I'll give it some more thought...

Reply:
Basically, `this._uiEditor` now also has the same function as `this._configValue.format` had before.

Reply:
By spreading it out over 2 properties, a bug could happen where they get out of sync. Putting it in 1 variable makes sure that no bug can cause this to happen.

Anyway, if you want to spend time on improving our editors, I'm not going to let this stand in your way. Both approaches can work just fine.

Reply:
No. I agree. As I said, I understand your point. It's definitely a bug waiting to happen.

That's actually why I had `_cardConfig` as a property before. Both `_cardConfig` and `_cardYaml`  started out as being properties of a custom class which kept track of the current state. Then I realized it could be optimized, but perhaps a bit too much...

Again, I'll keep thinking about this problem whether or not this is merged in the current state.